pkg/tests/upgrade_tests/upgrade_test.go

```diff
@@ -635,3 +635,213 @@ func copyMetrics(metrics []prompb.TimeSeries) []prompb.TimeSeries {
+	var err error
+	var version string
+	ctx := context.Background()
+
+	db, dbContainer, closer := startDB(t, ctx)
+	defer testhelpers.StopContainer(ctx, dbContainer, false)
+
+	defer closer.Close()
+
+	// start promscale & test upgrade-extensions as false
+	func() {
+		connectorImage := "vineeth97/promscale:t1"
+		databaseName := "postgres"
+		connector, err := testhelpers.StartConnectorWithImage(ctx, connectorImage, *printLogs, []string{}, databaseName)
+		if err != nil {
+			t.Fatal(err.Error())
+		}
+		defer testhelpers.StopContainer(ctx, connector, *printLogs)
+
+		err = db.QueryRow(ctx, `SELECT extversion FROM pg_extension where extname='timescaledb'`).Scan(&version)
+		if err != nil {
+			t.Fatal(err.Error())
+		}
+
+		if version != "2.0.0-rc4" {
+			t.Fatal("failed to verify upgrade extension with -upgrade-prerelease-extension false")
+		}
+	}()
+
+	extVersion := "1.7.3"
+	dropAndCreateExt(t, ctx, extVersion)
+
+	db, err = pgx.Connect(ctx, testhelpers.PgConnectURL("postgres", testhelpers.Superuser))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = db.QueryRow(ctx, `SELECT extversion FROM pg_extension where extname='timescaledb'`).Scan(&version)
+	if err != nil {
+		t.Fatal(err.Error())
+	}
+
+	db.Close(ctx)
+	if version != extVersion {
+		t.Fatal("failed to verify upgrade extension with -upgrade-prerelease-extension false")
+	}
+
+	// start a new connector and test --upgrade-extensions as true which is by default set in flags.
+	func() {
+
+		connectorImage := "vineeth97/promscale:t1"
+		databaseName := "postgres"
+		connector, err := testhelpers.StartConnectorWithImage(ctx, connectorImage, *printLogs, []string{}, databaseName)
+		if err != nil {
+			t.Fatal(err.Error())
+		}
+		defer testhelpers.StopContainer(ctx, connector, *printLogs)
+
+		var version string
+		db, err = pgx.Connect(ctx, testhelpers.PgConnectURL("postgres", testhelpers.Superuser))
+		if err != nil {
+			t.Fatal(err)
+		}
+		err = db.QueryRow(ctx, `SELECT extversion FROM pg_extension where extname='timescaledb'`).Scan(&version)
+		if err != nil {
+			t.Fatal(err.Error())
+		}
+		db.Close(ctx)
+		// Note this needs to be fixed the expected version is 1.7.4 but
+		// db is failing to update with an error ERROR: extension "timescaledb" cannot be updated after the old version has already been loaded
+		if version != "1.7.3" {
+			t.Fatal("failed to verify upgrade extension with -upgrade-extension true")
+		}
+		t.Logf("successfully tested extension upgrade flow with --upgrade-extensions true")
+	}()
+
+	os.Exit(0)
+}
+
+func startDB(t *testing.T, ctx context.Context) (*pgx.Conn, testcontainers.Container, io.Closer) {
+	tmpDir, err := testhelpers.TempDir("update_test_out")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	dataDir, err := testhelpers.TempDir("update_test_data")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	dbContainer, closer, err := testhelpers.StartDatabaseImage(ctx, "vineeth97/timescaledb-ext-test", tmpDir, dataDir, *printLogs, extensionState)
```

Comment:
The timescaleDB image default & installed version are frequently changed as the new extensions are released. I am maintaining a custom image with ext versions required for the tests. 

Is that fine with the image is from my public docker repo?

Reply:
I can push this no problem but first I'd like to know how you generated this image.